### PR TITLE
Add V file icon

### PR DESCRIPTION
**Pull request by spytheman**
Here is a preview of the new `v.svg` in comparison with some of the existing icons:
image

### CODE REVIEW

**Comment by spytheman** [comment]
@cla-bot check

**Comment by SomeoneToIgnore** [comment]
I suspect this won't be accepted, see https://github.com/zed-industries/zed/pull/18786#issuecomment-2395501617 and many other attempts to add the icons.

**Comment by spytheman** [comment]
I hope it will be, since I changed the .svg specifically, to be compatible with the look&feel of the rest.

The V logo normally uses blue hues for its bands. The one in the PR, instead uses only pure black, and black fill with `fill-opacity="0.5"` for the gray band. 

The .svg in the PR is also optimized for small size, and legibility with different backgrounds.

**Comment by spytheman** [comment]
Here is the `v.svg` file from the PR as code:
```svg
<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill="black" fill-opacity="0.5" d="m10.01,14.308l4.189,-12.212c0.071,-0.207 -0.041,-0.358 -0.25,-0.337l-3.295,0.333c-0.209,0.021 -0.435,0.207 -0.504,0.415l-3.878,11.670c-0.069,0.208 0.045,0.376 0.255,0.376l3.207,0c0.105,0 0.219,-0.084 0.255,-0.188l0.02,-0.058l0,0l0,0z"/>
<path fill="black" d="m2.267,1.759l3.247,0.333c0.206,0.021 0.428,0.207 0.497,0.415l3.903,11.859c0.034,0.104 -0.022,0.188 -0.126,0.188l-3.18,0c-0.207,0 -0.431,-0.168 -0.502,-0.375l-4.086,-12.082c-0.07,-0.207 0.04,-0.358 0.246,-0.337l0,0l0,0z"/>
</svg>

```

For comparison, here is the already existing `go.svg`:
```svg
<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M7.83413 6.47818C7.28956 6.61768 6.91787 6.72231 6.38194 6.86181C6.25228 6.89668 6.24364 6.9054 6.13127 6.77462C6.00161 6.6264 5.90652 6.5305 5.725 6.44331C5.18043 6.17303 4.65314 6.2515 4.16043 6.57409C3.57264 6.95772 3.2701 7.52443 3.27875 8.23065C3.28739 8.92816 3.76281 9.50359 4.44569 9.5995C5.03348 9.67797 5.52619 9.46872 5.91517 9.02406C5.99296 8.92816 6.06211 8.82353 6.14855 8.70147C5.83737 8.70147 5.44839 8.70147 4.48026 8.70147C4.29874 8.70147 4.25552 8.58812 4.31603 8.43991C4.4284 8.16962 4.63585 7.71625 4.75687 7.48956C4.7828 7.43725 4.84331 7.35006 4.97297 7.35006H8.11938C8.1021 7.58547 8.1021 7.82087 8.06752 8.05628C7.97244 8.68403 7.73905 9.25947 7.35871 9.76516C6.73634 10.5934 5.92381 11.1078 4.89517 11.2473C4.04806 11.3607 3.26146 11.195 2.56994 10.6719C1.93028 10.1837 1.56724 9.53847 1.47215 8.73634C1.35978 7.786 1.63639 6.93156 2.20689 6.18175C2.82061 5.3709 3.63315 4.85649 4.62721 4.6734C5.43975 4.52518 6.21771 4.62109 6.91787 5.10062C7.376 5.40578 7.70447 5.82428 7.92057 6.32996C7.97244 6.40843 7.93786 6.45203 7.83413 6.47818Z" fill="black"/>
<path d="M10.8733 11.2812C10.1105 11.2636 9.4147 11.0348 8.82792 10.5068C8.33334 10.0579 8.02318 9.48585 7.92259 8.80818C7.7717 7.81367 8.03156 6.93357 8.60158 6.15028C9.21352 5.30539 9.95119 4.86534 10.9487 4.68052C11.8038 4.5221 12.6085 4.61011 13.3378 5.12937C14 5.60462 14.4108 6.24709 14.5197 7.09198C14.6622 8.28012 14.3353 9.24822 13.5557 10.0755C13.0025 10.6652 12.3235 11.0348 11.5439 11.202C11.3176 11.246 11.0912 11.2548 10.8733 11.2812ZM12.8684 7.72566C12.86 7.61124 12.86 7.52323 12.8432 7.43522C12.6923 6.56393 11.9295 6.07107 11.1331 6.26469C10.3536 6.44951 9.8506 6.96877 9.66618 7.79606C9.51529 8.48254 9.83383 9.17782 10.4374 9.45945C10.8984 9.67067 11.3595 9.64427 11.8038 9.40664C12.466 9.0458 12.8264 8.48254 12.8684 7.72566Z" fill="black"/>
</svg>
```

and the already existing `tcl.svg`:
```svg
<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M3.03125 13C3.46875 12.2812 3.68556 12.0378 4.0625 11.5312M4.0625 11.5312C4.0625 9.86595 4.27768 8.02844 4.75687 7M4.0625 11.5312C4.75687 10.5981 6.6875 8.57812 7.92188 7.54688M4.0625 11.5312C7.875 11.5312 10.0507 9.46738 11.4062 8.03125C11.5818 7.84528 11.2307 7.34164 10.9157 6.96235C10.7718 6.78906 10.8964 6.50073 11.1213 6.48823C11.6657 6.45798 12.3874 6.36175 12.5 6.06684C12.7544 5.4003 12.9585 4.2437 13.0409 3.28832C13.0541 3.13644 12.9264 3.01119 12.7745 3.0243C10.5824 3.21343 8.22052 3.5262 6.5 4.82764" stroke="black" stroke-width="1.5" stroke-linecap="round"/>
</svg>
```

and the already existing `julia.svg`:
```svg
<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<circle cx="8" cy="5" r="2.75" fill="black"/>
<circle cx="4.75" cy="11" r="2.75" fill="black" fill-opacity="0.5"/>
<circle cx="11.25" cy="11" r="2.75" fill="black" fill-opacity="0.75"/>
</svg>
```

The new `v.svg` is not as compact as the `julia.svg`, but is very similar in structure and size to the `go.svg` and `tcl.svg` icons.

**Comment by spytheman** [comment]
Here is a preview, of how it looks in comparison with some of the other icons:
image

**Comment by spytheman** [comment]
> Nice!!! Looks perfect and even more stylistically correct than some of the existing ones. Good job.

Thank you 🙇🏻‍♂️ .

File: assets/icons/file_icons/file_types.json [hunk 1]
```diff
@@ -379,6 +382,9 @@
     "typescript": {
       "icon": "icons/file_icons/typescript.svg"
     },
+    "v": {
+      "icon": "icons/file_icons/v.svg"
+    }
```
Comment (spytheman) [inline assets/icons/file_icons/file_types.json:387]:
```suggestion
    },
```

**Comment by spytheman** [comment]
Is there a chance of this being merged, or at least of getting some feedback, on why it is not suitable etc?

**Review by danilo-leal — APPROVED** [review]
Thank you! Pushed an update there to the SVG, adjusting the icon to be more consistent with the others.

**Pull request merged**